sci-libs/cantera/cantera-2.4.0.ebuild

```diff
@@ -0,0 +1,144 @@
+
+inherit desktop python-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="http://www.cantera.org"
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+SRC_URI="https://github.com/Cantera/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+IUSE="+cpp +cti fortran pch python static-libs test"
+
+## Python2 is required by '<scons-3.0' to work.
+## Python3 automatic detection is used by cantera before compilling
+## if python3_package isn't set no "n".
+
+REQUIRED_USE="
+	|| ( cpp python )
+	cti? ( ${PYTHON_REQUIRED_USE} )
+	fortran? ( cpp )
+	python? ( cti )
+	static-libs? ( cpp )
+	?? ( python_targets_python3_4 python_targets_python3_5 python_targets_python3_6 python_targets_python3_7 )
+	"
+
+RDEPEND="
+	python? (
+		dev-python/numpy[${PYTHON_USEDEP}]
+	)
+"
+
+# SCons >= 3.0 is recomended, but builds with older versions
+DEPEND="
+	${RDEPEND}
+	dev-cpp/eigen
+	dev-libs/boost
+	dev-libs/libfmt:0=
+	dev-util/scons
+	sci-libs/sundials
+	fortran? (
+		sci-libs/sundials[fortran]
+		sys-devel/gcc[fortran]
+	)
+	python? (
+		dev-python/3to2
+		dev-python/cython[${PYTHON_USEDEP}]
+	)
+	test? (
+		>=dev-cpp/gtest-1.8.0
+	)
+"
+
+PATCHES=( "${FILESDIR}/${PN}_${PV}_disable_debug_and_optimization.patch" )
+
+## Full list of configuration options of Cantera is presented here:
+## http://cantera.org/docs/sphinx/html/compiling/config-options.html
+
+scons_vars=()
+set_scons_vars() {
+	scons_vars=(
+		CC="$(tc-getCC)"
+		CXX="$(tc-getCXX)"
+		cc_flags="${CXXFLAGS}"
+		cxx_flags="-std=c++11"
+		debug="no"
+		FORTRANFLAGS="${CXXFLAGS}"
+		use_pch=$(usex pch)
+## In some cases other order can break the detection of right location of Boost: ##
+		system_fmt="y"
+		system_sundials="y"
+		system_eigen="y"
+		extra_inc_dirs="/usr/include/eigen3"
+	)
+	use test || scons_vars+=( googletest="none" )
+}
+
+scons_targets=()
+set_scons_targets() {
+	scons_targets=(
+		prefix="/usr"
+		stage_dir="${D%/}"
+		f90_interface=$(usex fortran y n)
+	)
+
+	if use cti ; then
+		python_setup
+		local scons_python=$(usex python full minimal)
+		use python_targets_python2_7 && scons_targets+=( python2_package="${scons_python}" python2_cmd="python2.7" )
+		python_is_python3 && scons_targets+=( python3_package="${scons_python}" python3_cmd="${EPYTHON}" )
+
+		## Force setup of python{2,3}_package="none" if appropriate python_targets_python{2_7,3_x} isn't active
+		## regardless of USE 'cti' or/and 'python' are enabled
+		use python_targets_python2_7 || scons_targets+=( python2_package="none" )
+		python_is_python3 || scons_targets+=( python3_package="none" )
+	else
+		scons_targets+=( python2_package="none" python3_package="none" )
+	fi
+}
+
+src_compile() {
+	set_scons_targets
+	set_scons_vars
+	escons build "${scons_vars[@]}" "${scons_targets[@]}"
+}
+
+src_test() {
+	escons test
+}
+
+src_install() {
+	escons install
+	local lib_dirname=$(usex amd64 "lib64" "lib")
```

Comment:
`get_libdir`. Also you need to force this libdir on build system, not rely on it guessing the right directory.

Reply:
The Cantera SConstruct build script contains the rule
```
# Some distributions (e.g. Fedora/RHEL) use 'lib64' instead of 'lib' on 64-bit systems
if any(name.startswith('/usr/lib64/python') for name in sys.path):
    env['libdirname'] = 'lib64'
else:
    env['libdirname'] = 'lib'
```
If I properly understand the `get_libdir` from `multilib.eclass` return `lib64` or `lib32`. As I know the x86 distributive of Gentoo has only `lib` directory. Maybe then I should to patch SConstruct script by sed command within `scr_configure()` to always set the `env['libdirname']` to `lib` regardless of `amd64` or `x86` system? At least the `lib` is symlink to `lib64` on `amd64` systems. Is such approach is permissible then?

Reply:
Oh my. I hope they aren't using it to build Python install path (because it differs per version on Gentoo, and can't be guessed). In any case, I presume that `env['libdirname']` should indeed be forced somehow to match `$(get_libdir)`. Note that there are other values possible, e.g. `libx32` on x32 profile.

Reply:
I think it's common install path.

I asume then I have to add something like sed-patch to remove/comment some lines of conditional choose and to set `env['libdirname'] ` to `$(get_libdir)`

Reply:
Done.